core/modules/file/file.entity.inc

```diff
@@ -241,4 +241,94 @@ class FileStorageController extends EntityDatabaseStorageController {
       ->execute();
   }
 
+  /**
+   * Implements EntityControllerInterface::buildContent().
+   */
+  public function buildContent(EntityInterface $file, $view_mode = 'full', $langcode = NULL, $context = array()) {
+    if (!isset($langcode)) {
+      $langcode = $GLOBALS['language_content']->language;
+    }
+
+    // Remove previously built content, if exists.
+    $file->content = array();
+
+    // In case of a multiple view, file_view_multiple() already ran the
+    // 'prepare_view' step. An internal flag prevents the operation from running
+    // twice.
+    // Allow modules to change the view mode.
+    $view_mode = key(entity_view_mode_prepare('file', array($file->fid => $file), $view_mode, $langcode));
+    field_attach_prepare_view('file', array($file->fid => $file), $view_mode, $langcode);
+    entity_prepare_view('file', array($file->fid => $file), $langcode);
+
+    // Build the actual file display.
+    // @todo Figure out how to clean this crap up.
+    $file->content['file'] = file_view_file($file, $view_mode, $langcode);
+    if (isset($file->content['file'])) {
+      if (isset($file->content['file']['#theme']) && $file->content['file']['#theme'] != 'file_link') {
+        unset($file->content['file']['#file']);
+      }
+      unset($file->content['file']['#view_mode']);
+      unset($file->content['file']['#language']);
+    }
+    else {
+      unset($file->content['file']);
```

Comment:
All of this was moved over from `file_build_content` so that comment has been around for awhile. There's actually another issue to fix this https://github.com/backdrop/backdrop-issues/issues/5370. If it's merged we'll need to fix it here too.

Reply:
OK